src/synchronization_c.c4

```diff
@@ -183,20 +183,30 @@ SHMEM_BIND_C_SYNC(`SHMEM_DEF_WAIT_UNTIL')
 
 #define SHMEM_DEF_WAIT_UNTIL_ALL(STYPE,TYPE)                                                   \
     void SHMEM_FUNCTION_ATTRIBUTES                                                             \
-    shmemx_##STYPE##_wait_until_all(TYPE *vars, size_t nelems, int cond, TYPE value)           \
+    shmemx_##STYPE##_wait_until_all(TYPE *vars, size_t nelems,                                 \
+                                    int *status, int cond, TYPE value)                         \
     {                                                                                          \
         SHMEM_ERR_CHECK_INITIALIZED();                                                         \
         SHMEM_ERR_CHECK_SYMMETRIC(vars, sizeof(TYPE));                                         \
         SHMEM_ERR_CHECK_CMP_OP(cond);                                                          \
                                                                                                \
-        size_t i;                                                                              \
+        size_t i = 0, num_ignored = 0;                                                         \
                                                                                                \
-        if (nelems == 0) {                                                                     \
+        if (status) {                                                                          \
+            for (i = 0; i < nelems; i++) {                                                     \
+                if (status[i]) num_ignored++;                                                  \
+            }                                                                                  \
+        }                                                                                      \
+        if (nelems == 0 || num_ignored == nelems) {                                            \
```

Comment:
Do we have a unit test that exercises the `num_ignored == nelems` case?

Reply:
Good question. I don't think so.

Reply:
In such a unit test, we should just check whether `wait_until_all` returns, right? There is no need to check the completion of the AMO since the wait set is empty. 

Looking at the spec, there is one sentence at the end of the API description for wait_until_all which may not hold true in these cases.
```
Implementations must ensure that shmem_wait_until_all does not return before the update of the memory indicated by ivars is fully complete.
```
Do we have to add an unless clause indicating if the status array does not include all PEs.

Reply:
Doesn't this sentence cover the issue?

> If all elements in `status` are set to 1 or `nelems` is 0, the wait set is empty and this routine returns immediately.

Reply:
Yes, it does. But the last sentence in the API description is not covering the special case. I guess it was confusing to me since that line is written as a separate paragraph. It might be easier to merge this sentence to the end of the first paragraph.

Reply:
Ah, I see the confusion.  It might be correct, since there is no "update of the memory" in this special case right?  But yeah, it could use some clarification - if I understand correctly, your suggestion could be a doc edit.

Reply:
Yes, definitely doc edit should be sufficient. Similar edits may go to any/some/vector APIs as well.